src/storage/hp/primera/restapi/mode/volumeusage.pm

```diff
@@ -0,0 +1,174 @@
+                output_template => 'Used : %.2f %%',
+                perfdatas => [
+                    { template => '%.2f', min => 0, max => 100, unit => '%', label_extra_instance => 1, instance_use => 'name' }
+                ]
+            }
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+    
+    $options{options}->add_options(arguments => {
+        'filter-id:s'   => { name => 'filter_id' },
+        'filter-name:s' => { name => 'filter_name' }
+    });
+    
+    return $self;
+}
+
+sub prefix_volume_output {
+    my ($self, %options) = @_;
+    
+    return "Volume '" . $options{instance_value}->{name} . "' (#" . $options{instance_value}->{id} . ") ";
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $response = $options{custom}->request_api(endpoint => '/api/v1/volumes');
+    my $volumes = $response->{members};
+    $self->{volume} = {};
+
+    for my $volume (@{$volumes}) {
+        my $name = $volume->{name};
+        my $id   = $volume->{id};
+
+        if (defined($self->{option_results}->{filter_name}) and $self->{option_results}->{filter_name} ne '' and
+            $name !~ /$self->{option_results}->{filter_name}/) {
+            $self->{output}->output_add(long_msg => "Skipping volume named '" . $name . "': not matching filter /" . $self->{option_results}->{filter_name} . "/.", debug => 1);
+            next;
+        }
+        if (defined($self->{option_results}->{filter_id}) and $self->{option_results}->{filter_id} ne '' and
+            $id !~ /$self->{option_results}->{filter_id}/) {
+            $self->{output}->output_add(long_msg => "Skipping volume #" . $id . ": not matching filter /" . $self->{option_results}->{filter_id} . "/.", debug => 1);
+            next;
+        }
+
+        my $total     = $volume->{sizeMiB} * 1024 * 1024;
+        my $snap_used = $volume->{snapshotSpace}->{usedMiB};
+        my $adm_used  = $volume->{adminSpace}->{usedMiB};
+        my $usr_used  = $volume->{userSpace}->{usedMiB};
+
+        my $used = ($snap_used + $adm_used + $usr_used) * 1024 * 1024;
+        $self->{volume}->{$name} = {
+            id        => $id,
+            name      => $name,
+            total     => $total,
+            used      => $used,
+            free      => ($total - $used) >= 0 ? ($total - $used) : 0,
+            prct_used => $used * 100 / $total,
+            prct_free => (100 - ($used * 100 / $total) >= 0) ? (100 - ($used * 100 / $total)) : 0
+        };
+    }
+
+    if (scalar(keys %{$self->{volume}}) <= 0) {
+        $self->{output}->add_option_msg(short_msg => "No volume found.");
+        $self->{output}->option_exit();
+    }
+}
+
+1;
+
+__END__
+
+=head1 MODE
+
+Check volume usage.
+
+=over 8
+
+=item B<--filter-counters>
+
+Define which counters (filtered by regular expression) should be monitored.
```

Comment:
regex info should be the same format across all --filter* option (in this mode and the other)